esp32-wroom-rp/src/wifi.rs

```diff
@@ -73,10 +81,10 @@ where
         self.common.resolve(hostname)
     }
 
-    pub fn build_tcp_client(&'a mut self) -> TcpClient<S, C> {
-        TcpClient {
-            common: TcpClientCommon {
-                protocol_handler: &mut self.common.protocol_handler,
+    pub fn promote_to_transport(self) -> Tcp<'a, 'b,  S, C> {
+        Tcp {
+            inner: TcpInner {
```

Comment:
switched this to `inner` as we won't need a separate instance for SPI vs I2C since those concepts are hidden by `NinaProtocolHandler`